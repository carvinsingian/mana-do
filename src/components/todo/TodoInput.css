.ToDoInput__container {
  flex: 1 1;
}

.ToDoInput__input {
  width: 100%;
}
