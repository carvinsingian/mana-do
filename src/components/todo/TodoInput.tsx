import React, { useRef } from 'react'
import './TodoInput.css'

interface Props {
  onCreateTodo: (todo: string) => Promise<void>
}

const TodoInput = ({ onCreateTodo }: Props) => {
  const inputRef = useRef<HTMLInputElement>(null)

  const handleCreateTodo = (e: React.KeyboardEvent<HTMLInputElement>) => {
    //Ignore enter key when there is no value
    if (e.key === 'Enter' && inputRef.current && inputRef.current.value !== '') {
      onCreateTodo(inputRef.current.value)
      inputRef.current.value = '';
    }
  }

  return (
    <div className="ToDoInput__container">
      <input
        ref={inputRef}
        data-testid="todo-input"
        className="ToDoInput__input"
        placeholder="What needs to be done?"
        onKeyDown={handleCreateTodo}
      />
    </div>
  )
}

export default TodoInput