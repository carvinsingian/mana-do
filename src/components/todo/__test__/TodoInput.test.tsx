import React from 'react'
import ReactDOM from 'react-dom'
import TodoInput from '../TodoInput'
import { render, cleanup } from '@testing-library/react'

afterEach(cleanup)

test('Todo Input renders without crashing', () => {
  const div = document.createElement('div')
  const onCreateTodo = jest.fn()

  ReactDOM.render(<TodoInput onCreateTodo={onCreateTodo}/>, div)
})

test('Todo Input page matches snapshot', () => {
  const onCreateTodo = jest.fn()

  const { asFragment } = render(<TodoInput onCreateTodo={onCreateTodo}/>)
  const componentRender = asFragment()
  expect(componentRender).toMatchSnapshot()
})

test('Todo input is the page', () => {
  const onCreateTodo = jest.fn()

  const { getByTestId } = render(<TodoInput onCreateTodo={onCreateTodo} />)
  const todoInput = getByTestId('todo-input')

  expect(todoInput).toBeInTheDocument()
})
