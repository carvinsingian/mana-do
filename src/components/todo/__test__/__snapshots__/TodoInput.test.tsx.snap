// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo Input page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ToDoInput__container"
  >
    <input
      class="ToDoInput__input"
      data-testid="todo-input"
      placeholder="What needs to be done?"
    />
  </div>
</DocumentFragment>
`;
